test/Combinatorics/Matroids/Matroids.jl

```diff
@@ -137,8 +137,9 @@
 
         mg5 = dual_matroid(mg1)
```

Comment:
```suggestion
        mg5 = dual_matroid(cycle_matroid(g1))
```
The test needs that the input for the `dual_matroid` is allowed to be garbage-collected, e.g when it is a temporary object or when the input variable is reassigned `mg1 = nothing`. Maybe add a comment about this as well.
(Otherwise the test won't fail without this changeset)

Reply:
That makes perfect sense. I preferred the reassigning, and added a comment.